core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -35,16 +36,32 @@ object Entity {
 
   // The type parameter aids type inference in Message constructors. This should be unnecessary when Messages are covariant.
   def empty[F[_]]: Entity[F] = Empty
+  def strict(chunk: Chunk[Byte]): Entity[Pure] = Strict(chunk)
 
   final case class Default[+F[_]](body: EntityBody[F], length: Option[Long]) extends Entity[F] {
     def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1] = that match {
       case d: Default[F1] => Default(body ++ d.body, (length, d.length).mapN(_ + _))
+      case Strict(chunk) => Default(body ++ Stream.chunk(chunk), length.map(_ + chunk.size))
       case Empty => this
     }
 
     def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G] = Default(body.translate(fk), length)
   }
 
+  final case class Strict(chunk: Chunk[Byte]) extends Entity[Pure] {
```

Comment:
In @ChristopherDavenport's version that was a `case class Strict[F[_]](chunk: Chunk[Byte]) extends Entity[F]`, could you explain the reason for going with `Entity[Pure]` instead of `Entity[F]`. My intuition says this will cause some problems as at some point because `Client[IO]` and `Entity[Pure]` will collide.

Reply:
In this version `Entity` is covariant in `F`, so an `Entity[Pure]` is also an `Entity[IO]`. I don't expect this to cause any new problems given that `Entity` is already covariant, as is `Stream` which is what we currently use for `EntityBody`. Covariant `Entity` will also be needed for covariant `Message` per #5712.